vaadin-flow-components-shared-parent/vaadin-flow-components-base/src/main/java/com/vaadin/flow/component/shared/HasInputValue.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2000-2022 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.shared;
+
+import com.vaadin.flow.component.HasElement;
+import com.vaadin.flow.component.Synchronize;
+
+/**
+ * Mixin interface for components that has an internal input element. Intended
+ * for internal use.
+ *
+ * @author Vaadin Ltd
+ */
+public interface HasInputValue extends HasElement {
```

Comment:
I'm normally a big fan of interfaces and re-usability but I have the feeling that this (as stated in the web component) is a private API and should not be exposed as public methods and therefore not be an interface, instead I would suggest to add this method inside all components as protected of even private API so that flow users aren't bothered with it.

The name `hasInputValue` or `isInputValue` is so generic that a lot of people probably going to use it just to check if there is something inside the field, expecting that `setValue("something")` changes it to `true` even tho the web component states "input element has a non-empty value entered by the user".

Reply:
I have had similar fears. 

Hoped that **not** having type specified parameter type (unlike `HasValue`), documentation mentioned that it's `Intended for internal use` and not having a setter would discourage users from using this interface instead of the proper one. Initial naming was `isInputValuePopulated` which is probably more descriptive name in terms of what it is  returning.

However, I would agree that it still would be available for the users and somebody could probably use it by mistake.
I do appreciate your feedback as an actual user of our product! 🙇  I will change the approach in the PR in order to avoid confusions.

Reply:
I would argue that for most people any Java API is public that they can access, even tho there is Javadocs stating otherwise (which most people probably don't read) 

Like people typing "textField.value" in their IDE and use the first method that pop ups and works.. with this in mind moving this somewhere else sounds the best to reduce public API cluttering and improve DX. 

Another option could be to rename it so something totally technical like "isWebComponentUserPopulated" which sounds so obscure that probably less people are going to find and even use it.. but it perfectly describes it's meaning ;) But that feels like a way of "improved DX by obscurity" 😅